pkg/diagnose/connectivity/trace_request.go

```diff
@@ -0,0 +1,114 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package connectivity contains logic for connectivity troubleshooting between the Agent
+// and Datadog endpoints. It uses HTTP request to contact different endpoints and displays
+// some results depending on endpoints responses, if any.
+package connectivity
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/config/resolver"
+	"github.com/DataDog/datadog-agent/pkg/forwarder"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
+	"github.com/fatih/color"
+)
+
+// RunDatadogConnectivityDiagnose send requests to all known endpoints for all domains
+// to check if there are connectivity issues between Datadog and these endpoints
+func RunDatadogConnectivityDiagnose() error {
+	// Create domain resolvers
+	keysPerDomain, err := config.GetMultipleEndpoints()
+	if err != nil {
+		log.Error("Misconfiguration of agent endpoints: ", err)
+	}
+
+	// XXX: use NewDomainResolverWithMetricToVector ?
```

Comment:
Is it temporary code?

Reply:
It was a reminder but I put it as a note in the PR, I have removed it.